internal/controller/utils.go

```diff
@@ -237,12 +240,36 @@ func verifyRefreshInterval(interval string, allowedMinimalRefreshInterval time.D
 }
 
 func verifyWorkloadIdentityParameters(workloadIdentity *acpv1.WorkloadIdentityParameters) error {
-	if workloadIdentity.ManagedIdentityClientId == nil && workloadIdentity.ManagedIdentityClientIdReference == nil {
-		return loader.NewArgumentError("auth.workloadIdentity", fmt.Errorf("one of managedIdentityClientId and managedIdentityClientIdReference is required"))
+	if !strings.EqualFold(os.Getenv(WorkloadIdentityEnabled), "true") {
+		return loader.NewArgumentError("auth.workloadIdentity", fmt.Errorf("workloadIdentity is not enabled"))
 	}
 
-	if workloadIdentity.ManagedIdentityClientId != nil && workloadIdentity.ManagedIdentityClientIdReference != nil {
-		return loader.NewArgumentError("auth.workloadIdentity", fmt.Errorf("only one of managedIdentityClientId and managedIdentityClientIdReference is allowed"))
+	var authCount int = 0
+
+	if workloadIdentity.ManagedIdentityClientId != nil {
+		if strings.EqualFold(os.Getenv(WorkloadIdentityDisableGlobalServiceAccount), "true") {
```

Comment:
could you remind me again what's the issue to allow `ManageIdentityClientId` when global service account is disabled?

Reply:
The way of using `ManageIdentityClientId` is tied to the global service account, customer has the security concern that it cross the namespace security boundary, namespace owners can easily use the `ManageIdentityClientId` from other team(namespace) to access the App Config which they should not have the permission to access. 

This PR is aim to add the namespaced service account support to eliminate the security concern of customer. At the meanwhile, we give an option to disable the global service account to prevent them from using the risky way.

Reply:
I'm actually asking why the `ManageIdentityClientId` is tied to the global service account, instead of allowing it to be used by customer specified service account too.

Reply:
Are you saying why not support something like this?

```
auth:
   workloadIdentity:
      serviceAccountName: xxx
     ManageIdentityClientId: xxx
```

Reply:
yes. what's the reason behind?

Reply:
It will be ambiguous for users which one is exactly taking effect. If we allow this, there are some implicit assumptions involved. If both `serviceAccountName` and `managedIdentityClientId` are set, then managedIdentityClientId is assumed to be bound to the specified serviceAccountName. However, if serviceAccountName is not set, it may give the impression that serviceAccountName has a default value, which is the global service account. But this is not the case; namespaced service accounts and global service accounts are not the same and should not be conflated. This could easily lead to customer confusion.